src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -95,6 +96,33 @@ internal static IArgument CreateArgumentOperation(ArgumentKind kind, IParameterS
                  invokedAsExtensionMethod: invokedAsExtensionMethod);
         }
 
+        private ImmutableArray<IOperation> GetAnonymousObjectCreationInitializers(BoundAnonymousObjectCreationExpression expression)
+        {
+            // For error cases, the binder generates only the argument.
+            Debug.Assert(expression.Arguments.Length >= expression.Declarations.Length);
+
+            var builder = ArrayBuilder<IOperation>.GetInstance(expression.Arguments.Length);
+            for (int i = 0; i < expression.Arguments.Length; i++)
+            {
+                IOperation value = Create(expression.Arguments[i]);
+                if (i >= expression.Declarations.Length)
+                {
+                    builder.Add(value);
+                    continue;
+                }
+
+                IOperation target = Create(expression.Declarations[i]);
+                bool isInvalid = target.IsInvalid || value.IsInvalid;
```

Comment:
@heejaechang Is actually working on a PR to remove IOperation.IsInvalid flag. It is currently a very fragile heuristic, and we are replacing it with an extension method on IOperation that determines the result by computing if the span has syntax or semantic diagnostics within it.